src/procedures/__tests__/PushDividendPayment.ts

```diff
@@ -0,0 +1,312 @@
+        })
+      );
+    });
+
+    test('should throw if there is no valid dividend type being provided', async () => {
+      // Instantiate PushDividendPayment with incorrect dividend type
+      target = new PushDividendPayment(
+        { ...params, dividendType: 'wrong' as DividendType },
+        contextMock.getMockInstance()
+      );
+
+      await expect(target.prepareTransactions()).rejects.toThrow(
+        new PolymathError({
+          code: ErrorCode.ProcedureValidationError,
+          message: "Dividends of the specified type haven't been enabled",
+        })
+      );
+    });
+
+    test('should throw if an Erc20 dividend module is not attached', async () => {
+      target = new PushDividendPayment(
+        { ...params, dividendType: DividendType.Erc20 },
+        contextMock.getMockInstance()
+      );
+
+      wrappersMock.mock('getAttachedModules', Promise.resolve([]));
+
+      // Real call
+      await expect(target.prepareTransactions()).rejects.toThrowError(
+        new PolymathError({
+          code: ErrorCode.ProcedureValidationError,
+          message: "Dividends of the specified type haven't been enabled",
+        })
+      );
+    });
+
+    test('should throw if an Eth dividend module is not attached', async () => {
+      target = new PushDividendPayment(
+        { ...params, dividendType: DividendType.Eth },
+        contextMock.getMockInstance()
+      );
+
+      wrappersMock.mock('getAttachedModules', Promise.resolve([]));
+
+      // Real call
+      await expect(target.prepareTransactions()).rejects.toThrowError(
+        new PolymathError({
+          code: ErrorCode.ProcedureValidationError,
+          message: "Dividends of the specified type haven't been enabled",
+        })
+      );
+    });
+
+    test('should add a transaction to push a dividend payment of an attached ERC20 dividends module', async () => {
+      // Instantiate PushDividendPayment
+      target = new PushDividendPayment(
+        { ...params, dividendType: DividendType.Erc20 },
+        contextMock.getMockInstance()
+      );
+
+      wrappersMock.mock(
+        'getDividend',
+        Promise.resolve({
+          shareholders: [
+            { address: '0x01', paymentReceived: false },
+            { address: '0x02', paymentReceived: false },
+            { address: '0x03', paymentReceived: true },
+          ],
+        })
+      );
+
+      wrappersMock.mock(
+        'getAttachedModules',
+        Promise.resolve([erc20DividendsMock.getMockInstance()])
+      );
+
+      const addTransactionSpy = spy(target, 'addTransaction');
+
+      // Real call
+      await target.prepareTransactions();
+
+      // Verifications
+      expect(
+        addTransactionSpy
+          .getCall(0)
+          .calledWith(erc20DividendsMock.getMockInstance().pushDividendPaymentToAddresses)
+      ).toEqual(true);
+      expect(addTransactionSpy.callCount).toEqual(1);
```

Comment:
You are also missing one more line of code, which here would be
` expect(addTransactionSpy.getCall(0).lastArg.tag).toEqual(
        PolyTransactionTag.PushDividendPayment
      );`

See the function:
``` await this.addTransaction(dividendsModule!.pushDividendPaymentToAddresses, {
        tag: PolyTransactionTag.PushDividendPayment,
        // Only add resolver to the last transaction
        resolver:...
```
To see what we are getting at with this